float32 xRotation
float32 yRotation
float32 zRotation
